### PR TITLE
Plugins: Fix plugins CDN not working when plugins are not in domain's root path

### PR DESCRIPTION
**Pull request by xnyo**
**What is this feature?**

Plugins CDN path construction is broken if the plugins CDN root is in a "subfolder", ~~and if the path does not contain `plugin-cdn`.~~

This is what we use in production, and some files are not loading properly:

`https://grafana-assets.grafana.net/plugins/{id}/{version}/public/plugins/{id}/{assetPath}`

**Why do we need this feature?**

Fixes a bug in plugins CDN assets path construction.

**Who is this feature for?**

Hosted Grafana users

**Which issue(s) does this PR fix?**:

**Special notes for your reviewer**:

### CODE REVIEW

**Review by mckn — APPROVED** [review]
LGTM!

File: pkg/plugins/pluginscdn/pluginscdn.go [hunk 1]
```diff
@@ -3,15 +3,14 @@ package pluginscdn
...
+	"strings"
 
 	"github.com/grafana/grafana/pkg/plugins/config"
 )
 
 const (
-	// systemJSCDNKeyword is the path prefix used by system.js to identify the plugins CDN.
-	systemJSCDNKeyword = "plugin-cdn"
+	// systemJSCDNURLTemplate is a special path template used by system.js to identify plugin CDN assets
+	systemJSCDNURLTemplate = "plugin-cdn/{id}/{version}/public/plugins/{id}/{assetPath}"
```
Comment (wbrowne) [inline pkg/plugins/pluginscdn/pluginscdn.go:13]: Considering this is baked into the code this way, should the plugin CDN URL config value maybe instead be just the domain instead? 🤔

Comment (xnyo) [reply pkg/plugins/pluginscdn/pluginscdn.go:13]: Yeah I think it makes more sense and it'll be consistent. I'll change that! :+1:

File: pkg/plugins/pluginscdn/pluginscdn.go [hunk 2]
```diff
@@ -30,7 +31,7 @@ func ProvideService(cfg *config.Cfg) *Service {
 // and invalid base url.
 func (s *Service) NewCDNURLConstructor(pluginID, pluginVersion string) URLConstructor {
 	return URLConstructor{
-		cdnURLTemplate: s.cfg.PluginsCDNURLTemplate,
+		cdnURLTemplate: strings.TrimRight(s.cfg.PluginsCDNURLTemplate, "/") + "/" + cdnAssetPathTemplate,
```
Comment (xnyo) [inline pkg/plugins/pluginscdn/pluginscdn.go:34]: Config change:

old:
```
[plugins]
cdn_base_url = http://127.0.0.1:1234/plugins/{id}/{version}/public/plugins/{id}/{assetPath}
```

new:
```
[plugins]
cdn_base_url = http://127.0.0.1:1234/plugins
```

**Review by wbrowne — APPROVED** [review]
🚀

**Pull request merged**